packages/react-components/react-card/src/components/Card/Card.types.ts

```diff
@@ -51,9 +93,59 @@ export type CardProps = ComponentProps<CardSlots> & {
    * @default 'medium'
    */
   size?: 'small' | 'medium' | 'large';
+
+  /**
+   * Defines the controlled selected state of the card.
+   *
+   * @default false
+   */
+  selected?: boolean;
+
+  /**
+   * Defines whether the card is initially in a selected state or not when rendered.
+   *
+   * @default false
+   */
+  defaultSelected?: boolean;
+
+  /**
+   * Callback to be called when the selected state value changes.
+   */
+  onCardSelect?: (event: CardOnSelectEvent, data: CardOnSelectData) => void;
 };
 
 /**
  * State used in rendering Card.
  */
-export type CardState = ComponentState<CardSlots> & Required<Pick<CardProps, 'appearance' | 'orientation' | 'size'>>;
+export type CardState = ComponentState<CardSlots> &
+  Required<
+    Pick<CardProps, 'appearance' | 'orientation' | 'size'> & {
+      /**
+       * Represents a card that contains interactive events (MouseEvents) or is a button/a tag.
+       *
+       * @default false
+       */
+      isInteractive: boolean;
+
+      /**
+       * Represents a selectable card.
+       *
+       * @default false
+       */
+      isSelectable: boolean;
+
+      /**
+       * Represents a selectable card that contains a slot for the select element.
+       *
+       * @default false
+       */
+      hasSelectSlot: boolean;
```

Comment:
is this needed ? if the select slot is not needed a user could just pass `null` to the slot

Reply:
This is not a prop, but the state of the card after evaluating that it contains the select slot. We use it for styles to decide if we need to include styles to position the slot. :)